src/core/datastore/cachestore.js

```diff
@@ -99,12 +99,15 @@ export class CacheStore extends NetworkStore {
    * Pull items for the data store from the network to your local cache. A promise will be
    * returned that will be resolved with the result of the pull or rejected with an error.
    *
+   * IMPORTANT: This method is not intended to be used to make concurrent requests.
```

Comment:
This is about MLIBZ-2130. You may want to document this better, I think this is what we agreed upon in the comments of the ticket.